src/components/discovery/DiscoveryQueryBuilder.js

```diff
@@ -99,6 +103,7 @@ class DiscoveryQueryBuilder extends Component {
             (this.props.onSubmit ?? nop)();
         } catch (err) {
             console.error(err);
+            this.setState({isSubmitting: false});
```

Comment:
I think that we should add a `finally` statement in order to set isSubmitting back to false when no errors are caught.

Reply:
the `isSubmitting` signal is really one of the weakest parts of the code.... if it's immediately changed back will anything at all happen? I suspect it will get better performance in redux rather than as a prop.

Reply:
Yeah having it stored in redux would help with props drilling performance issues I think.
As far as having isSubmitting immediately set to false after a successful submit (whether with component of redux state), nothing will happen right now as far as I can see, but there could be side effects in the future if we rely on this state value.